daily_snapshot_terraform/modules/daily_snapshot/service/list_snapshots.rb

```diff
@@ -24,7 +26,7 @@ def list_snapshots(chain_name = 'calibnet', bucket = 'forest-snapshots', endpoin
       file.match(snapshot_format) { |m| 
         url = "https://#{bucket}.#{endpoint}/#{chain_name}/#{file}"
         url_s3 = "s3://#{bucket}/#{chain_name}/#{file}"
-        snapshot = Snapshot.new m[:network], m[:date].to_date, m[:height].to_i, m[:compressed], file, url_s3, url
+        snapshot = Snapshot.new m[:network], m[:date].to_date, m[:height].to_i, file, url_s3, url
```

Comment:
It might be worth making `Snapshot` a proper class to avoid such unexpected (`date.to_date`) conversions. It will be easier to extend it in the future as well. What do you think?

Reply:
I'm at the very edge of my Ruby skills. :) I struggle to grasp the differences between a struct and a class. Mind helping me convert the code to use a class?

Reply:
I find a Ruby `Struct` a shorthand for a simplified class, just when you want to combine some things. Kind of like a named tuple. Let me try to re-write it a bit.

Reply:
@lemmih I've pushed a proposition of how it might look like. Let me know what you think and please check if it works. It should but no promises.

Reply:
Nice, it works. Would it be okay to define `to_s` for `Snapshot`? Setting `to_s` to be the URL helped me check that it worked.

Reply:
Sure. Would you like me to add it or do you already have the code?

Reply:
I'll add it.

Reply:
Pushed.